src/hooks/useStyleTag.ts

```diff
@@ -14,7 +14,7 @@ import { CSSProperties } from '../types'
 export const useStyleTag = ({ styles, nodeId }: { styles: CSSProperties; nodeId?: string }) => {
   const [className, setClassName] = useState('')
 
-  useLayoutEffect(() => {
+  useEffect(() => {
```

Comment:
Wasn't there a reason we use `useLayoutEffect` here?

>> by checking `window !== typeof 'undefined' ?
>
> Unfortunately, we can not do that, because in that case react would complain due to "conditional use of hooks" which is prohibited

I think the correct way of doing that would be outside the component close to the import, basically

```
import { useLayoutEffect, useEffect } from 'react';

const useRightEffect = typeof window === 'undefined' ? useEffect : useLayoutEffect
```

and then within the component, you simply use the `useRightEffect`.

Reply:
I don't think that there was a reason to use `useLayoutEffect`. As Dimitry pointed out above, it was a suggestion from my side on top of Sofia's PR, because I thought it would better fit injecting the styles, but in the current situation - both will work the same way. The difference between useEffect and useLayoutEffect is not significant in order to try to save it this way.